### ISSUE TITLE
[v2.1.1-beta] Don't understand where tiddlers are saved (if saved)

### ISSUE THREAD

**Issue by mauloop** [Arlen22/TiddlyServer#55]
Hi Arlen,

it seems that I'm not able to save tiddlers or at least I don't understand if they are written somewhere (and where?)

My environment is made up by Windows 10 + Node.js 10.15.3 + Chrome 73. 

This is my config:

```
{
  "tree": {
    "$element": "group",
    "$children": {
      "NSlab": "O:\\github\\twlab\\NSlab\\wiki"
    }
  },
  "bindInfo": {
    "bindAddress": [
      "127.0.0.1"
    ],
    "filterBindAddress": false,
    "enableIPv6": false,
    "port": 27768,
    "bindWildcard": false,
    "hostLevelPermissions": {
      "localhost": {
        "mkdir": false,
        "upload": false,
        "settings": false,
        "WARNING_all_settings_WARNING": false
      },
      "*": {
        "mkdir": false,
        "upload": false,
        "settings": false,
        "WARNING_all_settings_WARNING": false
      }
    },
    "https": false,
    "_bindLocalhost": false
  },
  "logging": {
    "logColorsToFile": false,
    "logToConsoleAlso": false,
    "debugLevel": -1
  },
  "putsaver": {
    "etag": "",
    "etagWindow": 0,
    "backupDirectory": ""
  },
  "authAccounts": {},
  "directoryIndex": {
    "defaultType": "html",
    "icons": {
      "htmlfile": [
        "htm",
        "html"
      ]
    }
  },
  "EXPERIMENTAL_clientside_datafolders": {
    "enabled": false,
    "alwaysRefreshCache": true,
    "maxAge_tw_plugins": 31536000000
  },
  "authCookieAge": 2592000,
  "$schema": "./settings.schema.json"
}
```

NSlab is a brand new wiki (I started my test with an empty `tiddlers` folder). No plugins are loaded in `tiddlywiki.info`, but the _Tiddlyweb_ plugin.

Apparently there is nothing wrong while playing with my wiki. The server starts without errors, I can navigate to my wiki (http://localhost:27768/NSlab/) and if I make changes the "save" icon above the search box blinks as usual (grey-red-grey).

On the other hand, if I look at the `tiddlers` folder it is empty (no trace of the tiddlers that I created, nor the `$:/StoryList`).

If I refresh the page I can see my tiddlers in the _Recent_ tab, but the `tiddlers` folder is still empty.
If I restart the server then also the _Recent_ tab becames empty.

Here are some extracts from server stdout/stderr.

- Starting the server:

```
Settings file: K:\MK\opt64\TiddlyServer-2.1.1-beta\settings.json
$ref: keywords ignored in schema at path "#/additionalProperties"
TYPE: settings.bindInfo.https should be string
ADDITIONALPROPERTIES: settings.bindInfo.hostLevelPermissions['localhost'] should NOT have additional properties
ADDITIONALPROPERTIES: settings.bindInfo.hostLevelPermissions['localhost'] should NOT have additional properties
REQUIRED: settings.bindInfo.hostLevelPermissions['localhost'] should have required property 'writeErrors'
REQUIRED: settings.bindInfo.hostLevelPermissions['localhost'] should have required property 'websockets'
REQUIRED: settings.bindInfo.hostLevelPermissions['localhost'] should have required property 'registerNotice'
ADDITIONALPROPERTIES: settings.bindInfo.hostLevelPermissions['*'] should NOT have additional properties
ADDITIONALPROPERTIES: settings.bindInfo.hostLevelPermissions['*'] should NOT have additional properties
REQUIRED: settings.bindInfo.hostLevelPermissions['*'] should have required property 'writeErrors'
REQUIRED: settings.bindInfo.hostLevelPermissions['*'] should have required property 'websockets'
REQUIRED: settings.bindInfo.hostLevelPermissions['*'] should have required property 'registerNotice'
REQUIRED: settings.logging should have required property 'logAccess'
REQUIRED: settings.logging should have required property 'logError'
REQUIRED: settings.directoryIndex should have required property 'mixFolders'
32 32 64 16
 APP 2019-04-18 17:46:20 server 127.0.0.1 listening
Open your browser and type in one of the following:
http://127.0.0.1:27768
```

- Loading the wiki in Chrome:

```
[2019-04-18T17:47:04.396+-200] GET     127.0.0.1       200 127.0.0.1 /NSlab/                                                     6.260 ms - -
[2019-04-18T17:47:05.311+-200] GET     127.0.0.1       200 127.0.0.1 /NSlab/status                                               4.020 ms - -
[2019-04-18T17:47:05.329+-200] OPTIONS 127.0.0.1       404 127.0.0.1 /NSlab/                                                     3.099 ms - -
[2019-04-18T17:47:05.332+-200] HEAD    127.0.0.1       404 127.0.0.1 /NSlab/                                                     2.300 ms - -
[2019-04-18T17:47:06.444+-200] GET     127.0.0.1       200 127.0.0.1 /NSlab/recipes/default/tiddlers.json                        2.705 ms - -
[2019-04-18T17:47:06.847+-200] GET     127.0.0.1       200 127.0.0.1 /NSlab/favicon.ico                                          3.528 ms - -
[2019-04-18T17:47:07.841+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/%24%3A%2FStoryList          6.302 ms - -
```

- Creating a tiddler titled "test":

```
[2019-04-18T17:47:19.140+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/Draft%20of%20'New%20Tiddler' 4.707 ms - -
[2019-04-18T17:47:19.154+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/%24%3A%2FStoryList          2.387 ms - -
[2019-04-18T17:47:21.521+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/Draft%20of%20'New%20Tiddler' 5.266 ms - -
[2019-04-18T17:47:22.324+-200] DELETE  127.0.0.1       204 127.0.0.1 /NSlab/bags/default/tiddlers/Draft%20of%20'New%20Tiddler'   5.885 ms - -
[2019-04-18T17:47:23.360+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/%24%3A%2FStoryList          2.256 ms - -
[2019-04-18T17:47:23.379+-200] PUT     127.0.0.1       204 127.0.0.1 /NSlab/recipes/default/tiddlers/test                        3.544 ms - -
```

Browser console reports these errors:

```
$:/core/modules/utils/dom/http.js:71 OPTIONS http://localhost:27768/NSlab/ 404 (Not Found)
$:/core/modules/utils/dom/http.js:71 HEAD http://localhost:27768/NSlab/ 404 (Not Found)
```

I don't know what else I can check. Regards,

)+(auloop

**Comment by Marxsal** [comment]
Is it possible Windows is blocking that port? What happens if you try it on port 8080?

**Comment by mauloop** [comment]
I'm sure that the port works. If I switch back to TiddlyServer 2.0.4 using that port everything is ok. On the other side, I could not  connect to the server if the port was blocked by Windows.

**Comment by mauloop** [comment]
I've just found the cause of this behaviour. I removed the filesystem plugin from `tiddlywiki.info`. Once I've fixed this and the plugins was loaded everything work as expected and all the changed tiddlers are correctly saved in the `tiddlers` folder.

Error 404 for **OPTION** and **HEAD** requests are still there, but this is another story. I searched my old logs and found that they appeared when running v2.0.14 too. Apparently this have no consequences, but Arlen can decide to investigate it further.

Regards, )+(auloop
